packages/report-server/src/reportBuilder/transform/aliases/summaryAliases.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { Row } from '../../types';
+
+/**
+ * Inserts a new column with a value that is the calculation of the percentage of 1s out of 1s and 0s in a row.
+ * { rowFieldName: 'Availability of Male Condoms', facilityNameA: 'N', facilityNameB: 'N', facilityNameC: 'Y', facilityNameD: 'N'  }
+ *  => { rowFieldName: 'Availability of Male Condoms', facilityNameA: 'N', facilityNameB: 'N', facilityNameC: 'Y', facilityNameD: 'N', rowSummary: 0.75 }
+ */
+
+const detectColumnsToSummarise = (rows: Row[]): string[] => {
+  console.log(rows)
+  const anyRelevantColumns = rows.reduce((previousValue, currentValue) => {
+    const columnsWithYorN = Object.entries(currentValue).filter(([,value]) => value === 'Y' || value === 'N').map(([column,]) => column);
+    return previousValue.concat(columnsWithYorN);
+  },[] as string[])
+  const columnsToSummarise = anyRelevantColumns.reduce((previousValue: string[], currentValue: string): string[] => {
+    if (previousValue.indexOf(currentValue) > -1) {
+      return previousValue
+    }
+    previousValue.push(currentValue)
+    return previousValue
+  },[] as string[])
+
+  return columnsToSummarise
+};
+
+const addSummaryColumn = (row: Row, columnsToSummarise: string[]): Row => {
+  const numerator = Object.entries(row).filter(([key,value]) => columnsToSummarise.indexOf(key) > -1 && value === 'N').length
+  const denominator = Object.entries(row).filter(([key,]) => columnsToSummarise.indexOf(key) > -1).length
```

Comment:
```suggestion
  const denominator = Object.entries(row).filter(([column,value]) => columnsToSummarise.includes(column) && (value === 'Y' || value === 'N')).length
```
I think there may have been a bug here previously, what happens if the `value` was `undefined`?